tests/specs/functional/merkle-spec.k

```diff
@@ -14,40 +14,38 @@ endmodule
-          => doneMerkle( MerkleLeaf( .ByteArray, VALUE ) ) </k>
+    rule <k> runMerkle ( MerkleUpdate( .MerkleTree, .ByteArray, V ) )
+          => doneMerkle( MerkleLeaf( .ByteArray, V ) ) </k>
+      requires V =/=String ""
 
     // Update on MerkleLeaf
     rule <k> runMerkle ( MerkleUpdate( MerkleLeaf( #parseByteStack("0x0607"), _ ), #parseByteStack("0x0607"), V ) )
           => doneMerkle( MerkleLeaf ( #parseByteStack("0x0607"), V ) ) </k>
+      requires V =/=String ""
 
-    rule <k> runMerkle ( MerkleUpdate( MerkleLeaf( #parseByteStack("0x0607"), _ ), #parseByteStack("0x0608"), _ ) )
+    rule <k> runMerkle ( MerkleUpdate( MerkleLeaf( #parseByteStack("0x0607"), _ ), #parseByteStack("0x0608"), V ) )
           => doneMerkle( MerkleExtension( #parseByteStack("0x06"), ?_ ) ) </k>
+      requires V =/=String ""
 
-    rule <k> runMerkle ( MerkleUpdate( MerkleLeaf( #parseByteStack("0x05"), _ ), #parseByteStack("0x06"), _ ) )
+    rule <k> runMerkle ( MerkleUpdate( MerkleLeaf( #parseByteStack("0x05"), _ ), #parseByteStack("0x06"), V ) )
           => doneMerkle( MerkleBranch( ?_, ?_ ) ) </k>
+      requires V =/=String ""
 
     // Update on MerkleExtension
     rule <k> runMerkle ( MerkleUpdate( MerkleExtension( #parseByteStack("0x06"), .MerkleTree ), #parseByteStack("0x06"), V ) )
           => doneMerkle( MerkleExtension( #parseByteStack("0x06"), MerkleLeaf( .ByteArray, V ) ) ) </k>
+      requires V =/=String ""
 
-    rule <k> runMerkle ( MerkleUpdate( MerkleExtension( #parseByteStack("0x07"), _ ), #parseByteStack("0x06"), _ ) )
+    rule <k> runMerkle ( MerkleUpdate( MerkleExtension( #parseByteStack("0x07"), _ ), #parseByteStack("0x06"), V ) )
           => doneMerkle( MerkleBranch( ?_, ?_ ) ) </k>
+      requires V =/=String ""
 
-    rule <k> runMerkle ( MerkleUpdate( MerkleExtension( #parseByteStack("0x0708"), _ ), #parseByteStack("0x0709"), _ ) )
+    rule <k> runMerkle ( MerkleUpdate( MerkleExtension( #parseByteStack("0x0708"), _ ), #parseByteStack("0x0709"), V ) )
           => doneMerkle( MerkleExtension( #parseByteStack("0x07"), MerkleBranch( ?_, ?_ ) ) ) </k>
+      requires V =/=String ""
 
     // Update on MerkleBranch
     rule <k> runMerkle ( MerkleUpdate( MerkleBranch( M, _ ), .ByteArray, V ) )
           => doneMerkle( MerkleBranch( M, V ) ) </k>
-
-    rule <k> runMerkle ( .MerkleBranch )
-          => doneMerkle( MerkleBranch (  0 |-> .MerkleTree  1 |-> .MerkleTree  2 |-> .MerkleTree  3 |-> .MerkleTree
-                                         4 |-> .MerkleTree  5 |-> .MerkleTree  6 |-> .MerkleTree  7 |-> .MerkleTree
-                                         8 |-> .MerkleTree  9 |-> .MerkleTree 10 |-> .MerkleTree 11 |-> .MerkleTree
-                                        12 |-> .MerkleTree 13 |-> .MerkleTree 14 |-> .MerkleTree 15 |-> .MerkleTree
-                                      , ""
-                                      )
-                       )
-         </k>
+      requires V =/=String ""
```

Comment:
Can we combine all the proofs about merkle trees into a single file please? Just `tests/specs/functional/merkle-spec.k`. Just have different sections (in the same module `MERKLE-SPEC`) for each type of proof (putting, getting, deleting, etc...).